llvm/lib/CodeGen/MachineTraceMetrics.cpp

```diff
@@ -264,7 +257,7 @@ MachineTraceMetrics::Ensemble::
 getProcResourceDepths(unsigned MBBNum) const {
   unsigned PRKinds = MTM.SchedModel.getNumProcResourceKinds();
   assert((MBBNum+1) * PRKinds <= ProcResourceDepths.size());
-  return ArrayRef(ProcResourceDepths.data() + MBBNum * PRKinds, PRKinds);
+  return ArrayRef{ProcResourceDepths.data() + MBBNum * PRKinds, PRKinds};
```

Comment:
What is the reason this syntax is preferred?

Reply:
I think the syntax is newer, and is used to distinguish constructor calls from normal function calls.